tools/syz-declextract/clangtool/json.h

```diff
@@ -0,0 +1,78 @@
+// Copyright 2024 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+#ifndef SYZ_DECLEXTRACT_JSON_H
+#define SYZ_DECLEXTRACT_JSON_H
+
+#include <cassert>
+#include <cstdio>
+#include <memory>
+#include <string>
+#include <vector>
+
+class JSONPrinter {
+public:
+  JSONPrinter() : Top(*this) {}
+
+  template <typename T> void Field(const char* Name, const T& V, bool Last = false) {
+    printf("%s\"%s\": ", Indent(), Name);
+    print(*this, V);
+    printf("%s\n", Last ? "" : ",");
+  }
+
+  const char* Indent() const {
+    static std::string Indents;
+    while (Indents.size() < Nesting)
+      Indents.push_back('\t');
+    return Indents.c_str() + Indents.size() - Nesting;
+  }
+
+  class Scope {
+  public:
+    Scope(JSONPrinter& Printer, bool Array = false) : Printer(Printer), Array(Array) {
+      printf("%c\n", "{["[Array]);
+      Printer.Nesting++;
+      assert(Printer.Nesting < 1000);
+    }
+
+    ~Scope() {
+      assert(Printer.Nesting > 0);
+      Printer.Nesting--;
+      printf("%s%c", Printer.Indent(), "}]"[Array]);
+    }
+
+  private:
+    JSONPrinter& Printer;
+    const bool Array;
+  };
+
+private:
+  friend class Scope;
+  size_t Nesting = 0;
+  Scope Top;
+};
+
+inline void print(JSONPrinter& Printer, int V) { printf("%d", V); }
+inline void print(JSONPrinter& Printer, int64_t V) { printf("%ld", V); }
+inline void print(JSONPrinter& Printer, bool V) { printf("%s", V ? "true" : "false"); }
+inline void print(JSONPrinter& Printer, const char* V) { printf("\"%s\"", V ? V : ""); }
+inline void print(JSONPrinter& Printer, const std::string& V) { print(Printer, V.c_str()); }
```

Comment:
If it's ever required, it will fail gracefully. Go parser won't consume broken inputs.
I've already spent too much time on all of this, I would prefer to not spent more time right now on optional things, and complicate C++ code just in case.
Golden tests and Go parser are good enough guards.